src/components/MyAccount/NewSettings/EmailForm.tsx

```diff
@@ -0,0 +1,207 @@
+      )
+      setError(hasInvalidEmail)
+    }
+  }
+
+  const handleRemoveEmail = (index) => {
+    const updatedEmails = tempEmails.filter((_, i) => i !== index)
+    setTempEmails(updatedEmails)
+
+    // Immediately revalidate remaining emails.
+    const hasInvalidEmail = updatedEmails.some((email) => !validateEmail(email))
+    setError(hasInvalidEmail)
+  }
+
+  const handleAddEmail = () => {
+    const updatedEmails = [...tempEmails, ""]
+    setTempEmails(updatedEmails)
+
+    // Immediately revalidate all emails.
+    const hasInvalidEmail = updatedEmails.some((email) => !validateEmail(email))
+    setError(hasInvalidEmail)
+  }
+
+  const handleClearableCallback = (index) => {
+    const updatedEmails = [...tempEmails]
+    updatedEmails[index] = ""
+    setTempEmails(updatedEmails)
+    setError(true)
+  }
+
+  const cancelEditing = () => {
+    setTempEmails([...emails])
+    setIsEditing(false)
+    setError(false)
+  }
+
+  const submitEmails = async () => {
+    setIsLoading(true)
+    const validEmails = tempEmails.filter((email) => validateEmail(email))
+
+    try {
+      const response = await fetch(
+        `/research/research-catalog/api/account/settings/${patronData.id}`,
+        {
+          method: "PUT",
+          headers: {
+            "Content-Type": "application/json",
+          },
+          body: JSON.stringify({ emails: validEmails }),
+        }
+      )
+
+      if (response.status === 200) {
+        await getMostUpdatedSierraAccountData()
+        setIsSuccess(true)
+        setIsEditing(false)
+      } else {
+        setTempEmails([...emails])
+      }
+    } catch (error) {
+      console.error("Error submitting emails:", error)
+    } finally {
+      setIsLoading(false)
+    }
+  }
+
+  return (
+    <Flex flexDir="row" alignItems="flex-start" width="100%">
+      <Flex gap="xs" alignItems={"center"} sx={{ paddingTop: "xs" }}>
+        <Icon name="communicationEmail" size="medium" />
+        <Text
+          size="body1"
+          sx={{
+            fontWeight: "500",
+            width: "256px",
+            marginBottom: 0,
+          }}
+        >
+          Email
+        </Text>
+      </Flex>
+      {isEditing ? (
+        <Flex flexDir={"column"}>
+          {tempEmails.map((email, index) => (
+            <Flex key={index} mb="s">
+              <TextInput
+                name={`email-${index}`}
+                value={email}
+                id={`email-text-input-${index}`}
+                labelText="Update email"
+                showLabel={false}
+                isInvalid={error && !validateEmail(email)}
+                invalidText="Please enter a valid email address."
+                onChange={(e) => handleInputChange(e, index)}
+                isRequired
+                isClearable
+                isClearableCallback={() => handleClearableCallback(index)}
+                sx={{ width: "300px" }}
+              />
+              {index !== 0 && (
+                <Button
```

Comment:
Minorish but the focus ring floats over the input field:
<img width="102" alt="Screenshot 2024-10-10 at 3 18 57 PM" src="https://github.com/user-attachments/assets/9b98e8aa-24e7-479a-bce0-2983be698dcf">